### PR TITLE
Complex-valued ClientOnly annotation

### PR DESCRIPTION
**Pull request by HeikoTheissen**
As discussed with Evan and Bruce

### CODE REVIEW

**Review by ralfhandl — DISMISSED** [review]
This is cleaner than the two separate terms proposed in #204

File: vocabularies/Offline.xml [hunk 1]
```diff
@@ -49,8 +49,26 @@ Please refer to the [Offline OData product home](https://help.sap.com/doc/f53c64
            attempt to upload any changes that are made to them locally.
            The service as implemented on the server is described by the CSDL document if
            all elements tagged as `Offline.ClientOnly` are removed together with all subsequently dangling references.</String>
-        </Annotation>        
+        </Annotation>
       </Term>
+      <ComplexType Name="ClientOnlyType">
+        <Annotation Term="Common.Experimental" />
+        <Property Name="IsClientOnly" Type="Edm.Bool" Nullable="false" DefaultValue="true"/>
```
Comment (HeikoTheissen) [inline vocabularies/Offline.xml:56]: Can this property even be omitted? A `ClientOnly` annotation with an empty record as value would then mean "is client only".

Comment (ralfhandl) [reply vocabularies/Offline.xml:56]: Correct. An empty record without type specification means: all default values of the properties of the term's declared type.

Comment (HeikoTheissen) [reply vocabularies/Offline.xml:56]: I meant: Can the property be omitted from the complex type definition? That would syntactically rule out `"@Offline.ClientOnly": {"IsClientOnly": false, "ActiveEntitySet": "..."}`.

Comment (ralfhandl) [reply vocabularies/Offline.xml]: Yes, that's also possible, a complex type can have zero properties and only serve as a tagging type and base type.

I'd prefer that.

**Review by ralfhandl — APPROVED** [review]

File: vocabularies/Offline.xml (hunk 1 above)
Comment (HeikoTheissen) [reply vocabularies/Offline.xml]: > Yes, that's also possible, a complex type can have zero properties and only serve as a tagging type and base type.

But that leads to a wrong JSON representation, see oasis-tcs/odata-csdl-schemas#75.

**Review by ralfhandl — APPROVED** [review]

**Comment by HeikoTheissen** [comment]
Merged into #204.

**Pull request closed without merging**